tools/pipstrap.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+"""Uses pip to upgrade Python packaging tools to pinned versions."""
+from __future__ import absolute_import
+
+import os
+import shutil
+import tempfile
+
+import pip_install
+
+
+# We include the hashes of the packages here for extra verification of
+# the packages downloaded from PyPI. This is especially valuable in our
+# builds of Certbot that we ship to our users such as our Docker images.
+REQUIREMENTS = r"""
+pip==20.2.4 \
+    --hash=sha256:51f1c7514530bd5c145d8f13ed936ad6b8bfcb8cf74e10403d0890bc986f0033 \
+    --hash=sha256:85c99a857ea0fb0aedf23833d9be5c40cf253fe24443f0829c7b472e23c364a1
+setuptools==44.1.1 \
```

Comment:
Do you have a reason to not use the latest setuptools version available?

Reply:
[The next major version of `setuptools` after this one dropped support for Python 2](https://setuptools.readthedocs.io/en/latest/history.html#v45-0-0) and we're running `tools/pipstrap.py` with Python 2 for things like our unit tests with `tox`. If you try to install a newer version of `setuptools` here on Python 2, it won't work.

I personally think the fact that tests break if you try to update it may be sufficient, but I can add a comment about this tomorrow.

Reply:
Agree.